Zend/zend_vm_def.h

```diff
@@ -8133,7 +8133,7 @@ ZEND_VM_HANDLER(190, ZEND_COUNT, CONST|TMP|VAR|CV, UNUSED)
 			}
 
 			/* if not and the object implements Countable we call its count() method */
-			if (instanceof_function(Z_OBJCE_P(op1), zend_ce_countable)) {
+			if (zend_is_countable(op1)) {
```

Comment:
I get the same question at https://github.com/php/php-src/pull/3146 when trying to use `zend_is_iterable`. Isn't a good practice to DRY our code with functions? Or this is just for PHP, in C we need to focus on performance? :grimacing:

Reply:
The trouble is the code below (line 8139) expects the object to have a `count()` method, but `zend_is_countable()` will return true if an extension class defines a `count_elements` handler.

As this code is specifically calling the `count()` method, I think it should check `zend_ce_countable`

Reply:
@duncan3dc So, should I revert this changes, or revert the logic and give `zend_ce_countable` priority in the `switch` statement is also valid?

Reply:
@duncan3dc That point is a bit moot, since the checks performed in the code above will already have eliminated that possibility.

I do agree that this change should be reverted, though. Trying to reuse the logic in `zend_is_countable` isn't overly useful, since we've already duplicated similar checks in the code above.

Reply:
Reverting this changes as performance was affected :(